zenodo/modules/records/serializers/schemas/common.py

```diff
@@ -305,6 +315,26 @@ def validate_language(self, value):
                 field_name=['language']
             )
 
+    @validates('dates')
```

Comment:
It might be better to move this validation check in the `DateSchemaV1` class. This way, the REST API will return also exactly which items in the array (e.g. `dates[2]`) have an issue, instead of saying that the entire `dates` field has a problem

Reply:
I created this issue: https://github.com/zenodo/zenodo/issues/1749